internal/biz/model/resourcehistory.go

```diff
@@ -40,3 +41,21 @@ func (r *ResourceHistory) ResourceHistory(db *gorm.DB, s *schema.Schema) error {
 func (*ResourceHistory) TableName() string {
 	return "resource_history"
 }
+
+func (r *ResourceHistory) BeforeCreate(db *gorm.DB) error {
+	var err error
+	if r.ID == uuid.Nil {
+		r.ID, err = uuid.NewV7()
+		if err != nil {
+			return fmt.Errorf("failed to generate uuid: %w", err)
+		}
+	}
+
+	if r.ResourceId == uuid.Nil {
```

Comment:
Same thing for resource here as my other comment, I believe the resource should already exist and have an ID, and if not we may have a bigger issue.

Reply:
+1

Reply:
in this scenario, i think you are right. resource ID should be the UUID created for a resource (ID) and you can't have history on a resource without having a resource ID. Ill update this and give it a test, good call!